File: by-python/py-boj/1145-적어도 대부분의 배수.py
def solve(arr):
    maximum = arr[2] * arr[3] * arr[4]
    result = 0
    for k in range(arr[0], maximum + 1):
        count = 0
        for i in range(5):
            if k % arr[i] == 0 and k >= arr[i]:
                count += 1
            if count >= 3:
                result = k
                break
        else:
            continue
        break
    return result


if __name__ == '__main__':
    arr = list(map(int, input().split()))
    arr.sort()
    print(solve(arr))

File: by-kotlin/kt-boj/solutions/1145-적어도 대부분의 배수.kt
import java.util.*

fun main() = with(Scanner(System.`in`)) {
    val numbers = mutableListOf<Int>()
    repeat(5) { numbers.add(nextInt()) }

    numbers.sort()

    var result = 0
    val maximum = numbers[2] * numbers[3] * numbers[4]

    loop@ for (k in numbers[0]..maximum + 1) {
        var count = 0
        for (i in numbers.indices) {
            if (k % numbers[i] == 0 && k >= numbers[i]) {
                count += 1
            }
            if (count >= 3) {
                result = k
                break@loop
            }
        }
    }

    print(result)

}
